### PR TITLE
[WIP] Modularize code for MAX model

### PR DESCRIPTION
**Pull request by kmh4321**
- This PR is to modularize the [code](https://github.com/System-T/DimSim/blob/master/code/dimsim/dimsim/dimsim.py) from the original repo into different [modules](https://github.com/kmh4321/DimSim/tree/modular/dimsim) like `core`, `data` and `utils`.

- All the functions exposed directly to the user and immediate internal functions can be found in [`core`](https://github.com/kmh4321/DimSim/tree/modular/dimsim/core). The immediate internal functions have been prefixed with a `__` to indicate internal / private use.

- All internal functions (mainly pinyin converters and a few other non-essential functions) can be found in [`utils`](https://github.com/kmh4321/DimSim/tree/modular/dimsim/utils).

- The data pickle files can be found in [`data`](https://github.com/kmh4321/DimSim/tree/modular/dimsim/data).

- Test cases can be found under [`tests`](https://github.com/kmh4321/DimSim/tree/modular/dimsim/tests).

TODO: 

- [ ] finalze name of model 
- [ ] update readme 
- [ ] add code comments/docstrings 
- [ ] add more test cases (optional)
- [ ] add TravisCI 
- [ ] add exception handler

### CODE REVIEW

File: dimsim/tests/test_dimsim.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+def test_distance_near():
+        dist = get_distance('大侠','大虾')
+        assert dist == 0.0002380952380952381
+
+def test_distance_far():
+        dist = get_distance('大侠','大人')
+        assert dist == 25.001417183349876
+
+def test_get_candidates_simplified():
```
Comment (I-Hong) [inline dimsim/tests/test_dimsim.py]: Thanks @kmh4321 , I think we should add a test with a single `atom` for distance and candidates.

**Comment by frreiss** [comment]
@kmh4321 looks like you accidentally imported some MacOS `.DS_Store` files. You should remove those files from this branch an add ".DS_Store" to `.gitignore`.

**Comment by frreiss** [comment]
@kmh4321: Can you please update the the description for this PR with a list of high-level changes that are included in this change set? I can reverse-engineer most of the actions you performed by looking at the diff, but it would be easier to understand if there was an exhaustive list.

Also, what testing has been done on your branch?

**Comment by kmh4321** [comment]
@frreiss 

> @kmh4321 looks like you accidentally imported some MacOS `.DS_Store` files. You should remove those files from this branch an add ".DS_Store" to `.gitignore`.

I have already done this. The `.DS_Store` files in the git changelog are the deletions I made from the original version.




> @kmh4321: Can you please update the the description for this PR with a list of high-level changes that are included in this change set? I can reverse-engineer most of the actions you performed by looking at the diff, but it would be easier to understand if there was an exhaustive list.

Sure. I have basically broken down this [file](https://github.com/System-T/DimSim/blob/master/code/dimsim/dimsim/dimsim.py) from the original repo into different modules like [core](https://github.com/kmh4321/DimSim/tree/modular/dimsim/core), [data](https://github.com/kmh4321/DimSim/tree/modular/dimsim/data) and [utils](https://github.com/kmh4321/DimSim/tree/modular/dimsim/utils) in my branch/fork. I'll update the same in the PR description. There still isn't the code similar to other max models to raise exceptions/ return response codes for invalid inputs or other kinds of errors. I can add it on the TODO list.


> Also, what testing has been done on your branch?

I have some preliminary [tests](https://github.com/kmh4321/DimSim/tree/modular/dimsim/tests/test_dimsim.py) which run automatically during the build process. Need to setup travis to run it as well.

**Comment by kmh4321** [comment]
cc: @qk63030898 can you please review this at your convenience?

**Pull request merged**